### PR TITLE
Feature/profile picture

### PR DESCRIPTION
**Pull request by emonnin-epfl**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+import ch.epfl.sdp.healthplay.auth.ProfileActivity;
+import ch.epfl.sdp.healthplay.auth.SignedInActivity;
+import ch.epfl.sdp.healthplay.database.Database;
+import de.hdodenhof.circleimageview.CircleImageView;
+
+public class EditProfilePictureActivity extends AppCompatActivity {
+
+    private Uri uri;
+    private DatabaseReference mDatabase = FirebaseDatabase.getInstance(Database.DATABASE_URL).getReference();
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java:48]: You should probably make some of those field as final static fields.

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        ActivityResultLauncher<Intent> intent = registerForActivityResult(
+                new ActivityResultContracts.StartActivityForResult(), result ->{
+                    if(result.getResultCode() == Activity.RESULT_OK) {
+                        if(result.getData()!= null) {
+                            profilePictureView.setImageURI(result.getData().getData());
+                            uri = result.getData().getData();
+                        }
+                    }
+                    else {
+                        Toast.makeText(this, "try again!", Toast.LENGTH_SHORT).show();
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java:68]: You should make the toast String into a variable.

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        ProgressDialog dialog = new ProgressDialog(this);
+        dialog.setTitle("Set profile");
+        dialog.setMessage("Setting up profile");
+        dialog.show();
+
+        if(uri != null) {
+
+            FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+            if(user != null) {
+                StorageTask mtask = mStorage.child("profilePicture").child(user.getUid() + "jpg").putFile(uri);
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java:119]: Shouldn't there be a point before the "jpg", like rather ".jpg" ?

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                mtask.continueWithTask(new Continuation() {
+                   @Override
+                   public Object then(@NonNull Task task) throws Exception {
+                       if(!task.isSuccessful()) {
+                           throw task.getException();
+                       }
+                       else {
+                           return  mStorage.child("profilePicture").child(user.getUid() + "jpg").getDownloadUrl();
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java:127]: Same as above ? (Shouldn't there be a point before the "jpg", like rather ".jpg" ?)

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                    mUri = ((Uri)task.getResult()).toString();
+                    mDatabase.child(Database.USERS).child(mAuth.getCurrentUser().getUid()).child("image").setValue(mUri);
+                    dialog.dismiss();
+                }
+               });
+            }
+
+        }
+        else {
+            Toast.makeText(this, "Image not selected", Toast.LENGTH_SHORT).show();
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java:141]: You should make this string into a variable.

**Review by ozan-guven — CHANGES_REQUESTED** [review]
Good code in general, but look for the changes I asked. Furthermore, you should add some comments thoughout your code because it is sometimes difficult to understand what you are trying to do.

**Review by Antonio06188 — DISMISSED** [review]

File: app/src/main/java/ch/epfl/sdp/healthplay/auth/ProfileActivity.java [hunk 6]
```diff
@@ -29,14 +32,16 @@
 
 public class ProfileActivity extends AppCompatActivity {
 
-
+    private DatabaseReference mDatabase = FirebaseDatabase.getInstance(Database.DATABASE_URL).getReference();
```
Comment (Antonio06188) [inline app/src/main/java/ch/epfl/sdp/healthplay/auth/ProfileActivity.java:35]: There is an object Database which have mDatabase defines like this, and later in the program you create an Object Database for use the attribute mDatabase

File: app/src/main/java/ch/epfl/sdp/healthplay/auth/ProfileActivity.java [hunk 7]
```diff
@@ -29,14 +32,16 @@
...
+    private FirebaseAuth mAuth = FirebaseAuth.getInstance();
     @Override
     protected void onCreate(Bundle savedInstanceState) {
 
        SignedInActivity.SetMode(this);
 
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_profile);
+        getImage();
         FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
```
Comment (Antonio06188) [inline app/src/main/java/ch/epfl/sdp/healthplay/auth/ProfileActivity.java:45]: use mAuth

**Review by Antonio06188 — CHANGES_REQUESTED** [review]

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java (hunk 1 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java]: done!

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java (hunk 2 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java:71]: Did it!

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java (hunk 3 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java]: Yep you are right, thanks you.

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java (hunk 2 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java:71]: Probably not necessary, i use it only once.

File: app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java (hunk 5 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/EditProfilePictureActivity.java]: Did it!

File: app/src/main/java/ch/epfl/sdp/healthplay/auth/ProfileActivity.java (hunk 6 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/auth/ProfileActivity.java]: Yup i fixed it, thank you

File: app/src/main/java/ch/epfl/sdp/healthplay/auth/ProfileActivity.java (hunk 7 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/auth/ProfileActivity.java]: done!

**Review by ozan-guven — APPROVED** [review]
Thank you ! Keep up with work !

**Review by Antonio06188 — APPROVED** [review]

**Pull request merged**